README.md

```diff
@@ -53,7 +53,7 @@ Read the [overview](https://kueue.sigs.k8s.io/docs/overview/) and watch the Kueu
 
 ## Installation
 
-**Requires Kubernetes 1.25 or newer**.
+**Requires Kubernetes 1.29 or newer**.
```

Comment:
@mimowo Do we want to change this minimal required version?
I think that we can technically deploy kueue to v1.25 cluster.

Reply:
Yeah, I wouldn't like to couple the requirement bump to the k8s bump. 
Does it stop working on older versions @mbobrovskyi ?

Reply:
Reverted